Test/COD51010.TXT

```diff
@@ -0,0 +1,218 @@
+      DotNet_SymmetricAlgorithm.GetBlockSize(BlockSize);
+      Assert.AreEqual(128, BlockSize, 'SymmetricAlgorithm Get/Set Blocksize failed');
+    END;
+
+    [Test]
+    PROCEDURE TestSetGetKeySize@1000000009();
+    VAR
+      DotNet_SymmetricAlgorithm@1000000000 : Codeunit 50010;
+      KeySize@1000000001 : Integer;
+    BEGIN
+      Initialize();
+      DotNet_SymmetricAlgorithm.Create(Algorithm::Aes);
+      DotNet_SymmetricAlgorithm.SetKeySize(256);
+      DotNet_SymmetricAlgorithm.GetKeySize(KeySize);
+      Assert.AreEqual(256, KeySize, 'SymmetricAlgorithm Get/Set KeySize failed');
+    END;
+
+    [Test]
+    PROCEDURE TestSetGetFeedbackSize@1000000013();
+    VAR
+      DotNet_SymmetricAlgorithm@1000000000 : Codeunit 50010;
+      FeedbackSize@1000000001 : Integer;
+    BEGIN
+      Initialize();
+      DotNet_SymmetricAlgorithm.Create(Algorithm::Aes);
+      DotNet_SymmetricAlgorithm.SetBlockSize(128);
+      DotNet_SymmetricAlgorithm.SetFeedbackSize(64);
+      DotNet_SymmetricAlgorithm.GetFeedbackSize(FeedbackSize);
+      Assert.AreEqual(64, FeedbackSize, 'SymmetricAlgorithm Get/Set FeedbackSize failed');
+    END;
+
+    [Test]
+    PROCEDURE TestSetGetKey@1000000010();
+    VAR
+      DotNet_SymmetricAlgorithm@1000000000 : Codeunit 50010;
+      DotNet_Array@1000000001 : Codeunit 3000;
+      DotNet_ArrayKey@1000000002 : Codeunit 3000;
+      Index@1000000003 : Integer;
+    BEGIN
+      Initialize();
+
+      DotNet_Array.ByteArray(32);
+      FOR Index := 0 TO DotNet_Array.Length-1 DO
+        DotNet_Array.SetByteValue(Index, Index);
+
+      DotNet_SymmetricAlgorithm.Create(Algorithm::Aes);
+      DotNet_SymmetricAlgorithm.SetKeySize(256);
+      DotNet_SymmetricAlgorithm.SetKey(DotNet_Array);
+      DotNet_SymmetricAlgorithm.GetKey(DotNet_ArrayKey);
+      FOR Index := 0 TO DotNet_Array.Length-1 DO
+        Assert.AreEqual(
+          DotNet_Array.GetValueAsInteger(Index),
+          DotNet_ArrayKey.GetValueAsInteger(Index),
+          'SymmetricAlgorithm Get/Set Key failed');
+    END;
+
+    [Test]
+    PROCEDURE TestSetGetIV@1000000011();
+    VAR
+      DotNet_SymmetricAlgorithm@1000000000 : Codeunit 50010;
+      DotNet_Array@1000000003 : Codeunit 3000;
+      DotNet_ArrayIV@1000000002 : Codeunit 3000;
+      Index@1000000001 : Integer;
+    BEGIN
+      Initialize();
+
+      DotNet_Array.ByteArray(16);
+      FOR Index := 0 TO DotNet_Array.Length-1 DO
+        DotNet_Array.SetByteValue(Index, Index);
+
+      DotNet_SymmetricAlgorithm.Create(Algorithm::Aes);
+      DotNet_SymmetricAlgorithm.SetBlockSize(128);
+      DotNet_SymmetricAlgorithm.SetIV(DotNet_Array);
+      DotNet_SymmetricAlgorithm.GetIV(DotNet_ArrayIV);
+      FOR Index := 0 TO DotNet_Array.Length-1 DO
+        Assert.AreEqual(
+          DotNet_Array.GetValueAsInteger(Index),
+          DotNet_ArrayIV.GetValueAsInteger(Index),
+          'SymmetricAlgorithm Get/Set IV failed');
+    END;
+
+    PROCEDURE SymmetricAlgorithmCreateAes@1000000000();
```

Comment:
This should be a separate test case

Reply:
Please confirm.
I am interpretting this comment to mean - TestSetGetIV should be separated into two tests - TestSetIV & TestGetIV.
I assume this comment also applies to the other "combined" tests:
TestSetGetMode, TestSetGetPadding, TestSetGetBlockSize, TestSetGetKeySize, TestSetGetFeedbackSize, TestSetGetKey

Reply:
What I meant is that `TestSymmetricAlgorithmCreate` shouldn't be testing the creation of various algorithms. Instead, every `SymmetricAlgorithmCreate*` should be a separate test case.